Feature/first 34

rlarlgkd opened the pull request:
기기 + 알림 기본 api 작성

Commit:
Device 도메인 CRUD 세팅(컨트롤러 및 서비스는 아직 작업 중)

Commit:
Device 도메인 CRUD 세팅(컨트롤러 및 서비스는 아직 작업 중)

rlarlgkd committed:
global -> exception 추가

Commit:
캐멀케이스로 엔티티 변수 변경

Commit:
기기 + 알림 기본 CRUD 완성

Commit:
Merge remote-tracking branch 'origin/feature/FIRST-34' into feature/FIRST-34

# Conflicts:
#	src/main/java/com/heygongc/device/application/DeviceService.java
#	src/main/java/com/heygongc/device/application/DeviceServiceImpl.java
#	src/main/java/com/heygongc/device/domain/Device.java
#	src/main/java/com/heygongc/device/domain/DeviceRepository.java
#	src/main/java/com/heygongc/device/presentation/DeviceController.java
#	src/main/java/com/heygongc/device/presentation/request/DeviceNameRequest.java
#	src/main/java/com/heygongc/device/presentation/response/DeviceResponse.java

rlarlgkd committed:
Merge branch 'develop' into feature/FIRST-34

# Conflicts:
#	src/main/java/com/heygongc/user/domain/User.java

rlarlgkd committed:
디바이스 스웨거 + DTO 정리

rlarlgkd committed:
디바이스 익셉션 추가

rlarlgkd committed:
디바이스 + 알림 도메인의 익셉션, DTO, Swagger, QueryDSL 부분 수정.

sbywww commented on src/main/java/com/heygongc/device/domain/CustomDeviceRepositoryImpl.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    @Override
+    public Device findDeviceBySeqAndUser(Long deviceSeq, User user) {
```
메소드명을 조금 더 명확하게 findMyDevice(Long deviceSeq, User user)로 하는게 어떨까요?

sbywww commented on src/main/java/com/heygongc/device/application/DeviceService.java:
```diff
@@ -64,47 +58,39 @@ public DeviceResponse addDevice(User user, DeviceInfoRequest request) {
-                .orElseThrow(() -> new DeviceNotFoundException());
+    @Transactional
+    public Device updateDevice(Long deviceSeq, String deviceName, User user) {
+        Device device = deviceRepository.findDeviceBySeqAndUser(deviceSeq, user);
+        if (device == null) {
+            throw new DeviceNotFoundException();
+        }
```
반환값을 Optional로 처리하는게 일관성 있는 코드가 될 것 같습니다.

sbywww commented on src/main/java/com/heygongc/device/application/DeviceService.java:
```diff
@@ -64,47 +58,39 @@ public DeviceResponse addDevice(User user, DeviceInfoRequest request) {
-            );
-
-            return deviceResponse;
+            // Changes to 'device' are automatically tracked and persisted by JPA at transaction completion
```
deviceName이 null인 상황이 있을까요?
만약 필수 값이라고 하면, controller에서 validate로 처리하는게 좋을 것 같습니다.

추가로 setter를 지양하고 device.changeDeviceInfo(deviceName); 이렇게 처리하는 방법이 더 좋을 것 같습니다.

rlarlgkd replied:
디바이스 널인 상황 없습니다. 네 valid어노테이션 넣고 셋네임 대신 해당 메서드 새로 만들어서 처리하였습니다!

sbywww commented on src/main/java/com/heygongc/device/application/DeviceService.java:
```diff
@@ -26,27 +26,21 @@ public DeviceService(DeviceRepository deviceRepository, UserRepository userRepos
+
         String[] parts = request.deviceQR().split("_");
         Long parsedDeviceSeq = Long.parseLong(parts[0]);
         String type = parts[1];
```
해당코드는 DeviceInfoRequest에서 처리하는게 좋아보일 것 같습니다.

rlarlgkd committed:
Optional 사용하여 익셉션 코드 단축(notification 이랑 device)

setter 사용 지양! → 보안에도 더 좋고 비즈니스 로직도 명확해진다! (notification + device 둘 다 업데이트 api 에서 세터 대신 새로운 메서드를 만들어줬다)

DTO 로직은 되도록이면 DTO 안에서 처리 할 수 있도록 변경!

조금 불명확한 메서드명 좀 더 명확하게 변경.

Zeeeen0 commented on src/main/java/com/heygongc/notification/application/NotificationService.java:
```diff
@@ -0,0 +1,83 @@
+                List<Notification> toDelete = notifications.subList(100, notifications.size());
+                notificationRepository.deleteAll(toDelete);
+            }
+        }
```
findList해서 100개 초과 데이터를 subList로 만들어서 delete하는 방식인 것 같은데,
사용하지 않는 List를 만들게 되어 메모리/성능 이슈가 있을 수도 있지 않을까 싶어요

count한 후 100번째 이후 데이터를 delete하면 어떨까요?

Zeeeen0 commented on src/main/java/com/heygongc/device/domain/Device.java:
```diff
@@ -0,0 +1,62 @@
+    @OneToMany(mappedBy = "notification", cascade = CascadeType.ALL, orphanRemoval = true)
+    @Builder.Default
+    private List<Notification> notifications = new ArrayList<>();
+    public void changeDeviceInfo(String deviceName) {
```
메소드명에 Info보단 Name이 맞지 않을까 싶습니다 !

Zeeeen0 commented on src/main/java/com/heygongc/device/presentation/DeviceController.java:
```diff
@@ -0,0 +1,157 @@
+    public ResponseEntity<ListResponse<Device>> getAllDevice(@Parameter(hidden = true) @LoginUser User user){
+        Long userSeq = user.getSeq();
+        List<Device> devices = deviceService.getAllDevices(userSeq);
+        ListResponse<Device> deviceList = new ListResponse<Device>(
+                devices
+        );
```
User, Notification도 같이 리턴되지 않나요 ??
DeviceResponse를 사용하지 않은 이유가 궁금합니다!

rlarlgkd committed:
ListResponse 삭제
deleteOldNotification 수정
changeDeviceInfo -> changeDeviceName 으로 메서드 변경

rlarlgkd committed:
deleteOldNotification 삭제
